packages/core/user-profile/core-user-profile-browser/src/service.ts

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { Observable } from 'rxjs';
+import type { AuthenticatedUser } from '@kbn/core-security-common';
+import type {
+  UserProfileData,
+  UserProfileWithSecurity,
+  UserProfile,
+} from '@kbn/core-user-profile-common';
+
+export interface UserProfileService {
+  /**
+   * Retrieve an observable emitting when the user profile is loaded.
+   */
+  getUserProfile$(): Observable<UserProfileData | null>;
+
+  /**
+   * Retrieves the user profile of the current user. If the profile isn't available, e.g. for the anonymous users or
+   * users authenticated via authenticating proxies, the `null` value is returned.
+   * @param [params] Get current user profile operation parameters.
+   * @param params.dataPath By default `getCurrent()` returns user information, but does not return any user data. The
+   * optional "dataPath" parameter can be used to return personal data for this user.
+   */
+  getCurrent<D extends UserProfileData>(
+    params?: UserProfileGetCurrentParams
+  ): Promise<GetUserProfileResponse<D>>;
```

Comment:
TSDoc was just copied as it was from the security types. But I don't think we need to change anything on it?